frontend/src/composables/useShootEditor/index.js

```diff
@@ -295,40 +341,38 @@ export function useShootEditor (initialValue, options = {}) {
   }
 
   function execUndo () {
-    if (cm.value) {
-      cm.value.execCommand('undo')
-      cm.value.focus()
+    if (cmView.value) {
+      undo(cmView.value)
     }
   }
 
   function execRedo () {
-    if (cm.value) {
-      cm.value.execCommand('redo')
-      cm.value.focus()
+    if (cmView.value) {
+      redo(cmView.value)
     }
   }
 
   watchEffect(() => {
-    if (cm.value && schemaDefinition.value) {
+    if (cmView.value && schemaDefinition.value) {
       const shootProperties = get(schemaDefinition.value, 'properties', {})
       completions.value = new EditorCompletions(shootProperties, {
-        cm: cm.value,
+        cmView: cmView.value,
         completionPaths: get(options, 'completionPaths', []),
         logger,
       })
     }
   })
 
-  watch(isReadOnly, value => {
-    if (cm.value) {
-      cm.value.setOption('readOnly', value)
-    }
+  watch(isReadOnly, isReadOnly => {
+    cmView.value.dispatch({
+      effects: readonlyCompartment.reconfigure(EditorView.editable.of(isReadOnly)),
+    })
   })
 
-  watch(cmTheme, value => {
-    if (cm.value) {
-      cm.value.setOption('theme', value)
-    }
+  watch(isDarkMode, isDarkMode => {
+    cmView.value.dispatch({
+      effects: themeCompartment.reconfigure([isDarkMode ? oneDark : [], syntaxHighlighting(isDarkMode ? oneDarkHighlightStyle : defaultHighlightStyle)]),
+    })
   })
```

Comment:
added additional line breaks for better readability
```suggestion
  watch(isDarkMode, isDarkMode => {
    cmView.value.dispatch({
      effects: themeCompartment.reconfigure([
        isDarkMode ? oneDark : [],
        syntaxHighlighting(isDarkMode ? oneDarkHighlightStyle : defaultHighlightStyle)
      ])
    })
  })
```